### DISCUSSION TITLE
Data set test Adresses with a big file: Test run fails permanetly and remains in status undefined

### DISCUSSION THREAD

**Discussion by g-weber** [INSPIRE-MIF/helpdesk-validator#1005] [Q&A] [answered]
Dear all
Multiple attempts to run a test with Conformace Classes for data set Adresses always failed when using the attached ZIP-file as test ressource.
After longer runtimes it always got stuck and remains in  status "UNDEFINED". See attached logfile as an example 
[ad_4258.zip](https://github.com/INSPIRE-MIF/helpdesk-validator/files/8388937/ad_4258.zip)
[LOG_Test run on 14_13 - 29_03_2022 with test suite Annex I - Addresses.txt](https://github.com/INSPIRE-MIF/helpdesk-validator/files/8388954/LOG_Test.run.on.14_13.-.29_03_2022.with.test.suite.Annex.I.-.Addresses.txt)

**Comment by fabiovinci** [comment]
Dear @g-weber,

I had a look at your gml and it is very large, for this reason it takes a long time and the status remains "UNDEFINED" but the test should stay running.
In the meantime, it is better if you don't start other tests in order to don't overload the validator.

Looking at the log of the current validation (https://inspire.ec.europa.eu/validator/test-reports/details.html?id=EID1a273743-870c-4459-abb9-b14fc41b1694) it seems it’s running, but there is also the possibility that the validation cannot be completed due to the large file size.

I saw that there are some internal reference in your gml and for each of them the validator verifies that the corrosponding feature type is present in the dataset itself. So, for a file of 1300 MB is quite hard.

![image](https://user-images.githubusercontent.com/22212406/161113219-ca9f1aec-4c93-4fb7-87e7-c5f337cfcd63.png)

**Comment by g-weber** [comment]
Dear @fabiovinci,

Thanks for your initial feedback. Regarding your comment about the size of the gml file:
Are there actually any defined or known limits on the maximum file size for a gml file that the validator can handle?

**Comment by fabiovinci** [comment]
Dear @g-weber,

a limit on the maximum file size is difficult to define since it depends on many factors, i.e. complexity of the GML structure, numbers of points defining the geometry, feature type references.

Looking at the logs of the different attempts to validate the gml file it seems that they all stop at the same point:

![image](https://user-images.githubusercontent.com/22212406/161289304-883bfa38-d854-4bbf-9091-8e953312edeb.png)

Since the next test is "gmlas.d.10: Validate geometries", I think it will require many time and it is possible that the validator will not be able to complete it.

**Comment by fabiovinci** [comment, marked as answer]
Dear @g-weber,

I can suggest either installing a local version of the Validator and trying to validate locally your dataset or creating a significant sample data and validating it with the production instance.
